mllib/src/main/scala/org/apache/spark/ml/tuning/ParamGridBuilder.scala

```diff
@@ -118,3 +118,63 @@ class ParamGridBuilder @Since("1.2.0") {
     paramMaps
   }
 }
+
+
+object ParamGridBuilder {
+
+  /**
+   * Split the given list of paramMaps in two such that the left list contains the supplied
+   * params and the right list does not.
+   *
+   * @param paramMaps Array of paramMaps to split
+   * @param params Array of params that will be in the left list
+   * @return Tuple of two Array[ParamMap]
+   */
+  def splitOnParams(paramMaps: Array[ParamMap],
+                    params: Array[Param[_]]): (Array[ParamMap], Array[ParamMap]) = {
+    val leftValues = mutable.Map.empty[Param[_], mutable.LinkedHashSet[Any]]
```

Comment:
This could be done more efficiently, I just went with a simple routine that I knew would generate the correct output.